.github/workflows/integration_test_run.yaml

```diff
@@ -204,6 +208,17 @@ jobs:
       - name: Pre-run script
         if: ${{ inputs.pre-run-script != '' }}
         run: bash ${{ inputs.pre-run-script }}
+      - uses: actions/cache@v3
+        id: cache-integration
+        with:
+          path: |
+            '**/*.charm'
+            .tox
+          key: integration-cache-${{ steps.get-date.outputs.date }}-${{ hashFiles('**/charmcraft.yaml', '**/metadata.yaml', '**/actions.yaml', '**/config.yaml', '**/src') }}
```

Comment:
Any particular reason for including the date? Is it so that dependencies are re-installed frequently, eg., if there is a new patch version available?

Reply:
The date is to ensure the cache gets invalidated at some point, in the particular case it lasts at most one week. The reason behind it is that not all dependencies we have are pinned to specific versions and we don't bump them in the files that are hashed for the cache key, so there needs to be a safety mechanism to invalidate the cache at some point, otherwise it will get stale after a while.